pkgs/development/ocaml-modules/pprint/default.nix

```diff
@@ -1,38 +1,20 @@
-{ lib, stdenv, fetchurl, ocaml, findlib, ocamlbuild }:
+{ lib, fetchurl, buildDunePackage }:
 
-assert lib.versionAtLeast (lib.getVersion ocaml) "3.12";
-
-let param =
-  if lib.versionAtLeast ocaml.version "4.02"
-  then {
-    version = "20171003";
-    sha256 = "06zwsskri8kaqjdszj9360nf36zvwh886xwf033aija8c9k4w6cx";
-  } else {
-    version = "20140424";
-    sha256 = "0sc9q89dnyarcg24czyhr6ams0ylqvia3745s6rfwd2nldpygsdk";
-}; in
-
-stdenv.mkDerivation {
-
-  name = "ocaml${ocaml.version}-pprint-${param.version}";
+buildDunePackage rec {
+  pname = "pprint";
+  version = "20200410";
 
   src = fetchurl {
-    url = "http://gallium.inria.fr/~fpottier/pprint/pprint-${param.version}.tar.gz";
-    inherit (param) sha256;
+    url = "https://github.com/fpottier/pprint/archive/refs/tags/${version}.tar.gz";
```

Comment:
Please use `fetchzip` (or `fetchFromGitHub`) for this kind of URL.

Reply:
Done.

Reply:
1) Avoid `fetchFromGitHub, fetchFromGitLab` if there is a release archive.
2) .tar.gz => fetchzip, fetchFromGitHub, fetchFromGitLab
3) .tbz => fetchurl
Is there some other consideration?

Reply:
This kind of question is better asked on the [forum](https://discourse.nixos.org/).

First consideration: where is the source code for a particular version? Sometimes it is different from what is tracked in the VCS. This impact the choice of the URL.

Second consideration: will two downloads at the same URL return the same data? If it is the case, then use `fetchurl`, otherwise `fetchzip` & co.